### PR TITLE
Fix parameters for Snowflake create tasks statement

### PR DESCRIPTION
**Pull request by chwiese**
### Brief summary of the change made

Fixes options for the CREATE TASK statement in the Snowflake dialect:
- One can only specify the `WAREHOUSE` parameter _or_ the `USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE` parameter, but not both
- The parameters can be in arbitrary order, except for `AFTER` which must be last
- The value for `USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE` is a warehouse size, not a quoted literal. Warehouse sizes themselves are not identifiers, so I am changing the type of that RegexParser.
- The quotes around the warehouse sizes are optional, except for the ones with hyphens, which must be quoted. The same rules apply the `CREATE WAREHOUSE` statements, even if that is not clearly documented there.

### Are there any other side effects of this change that we should be aware of?

None.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 1]
```diff
@@ -1653,55 +1654,52 @@ class CreateTaskSegment(BaseSegment):
...
+                Sequence(
+                    Ref("ParameterNameSegment"),
+                    Ref("EqualsSegment"),
+                    OneOf(
+                        Ref("BooleanLiteralGrammar"),
+                        Ref("QuotedLiteralSegment"),
+                        Ref("NumericLiteralSegment"),
+                    ),
                 ),
+                delimiter=Ref("CommaSegment"),
```
Comment (jpy-git) [inline src/sqlfluff/dialects/dialect_snowflake.py:1690]: This is the default delimiter so we don't need it here

**Review by jpy-git — COMMENTED** [review]
@chwiese very small tweak to remove the unnecessary default delimiter, but otherwise perfect and will approve and merge after once that line is removed ⭐

File: src/sqlfluff/dialects/dialect_snowflake.py (hunk 1 above)
Comment (chwiese) [reply src/sqlfluff/dialects/dialect_snowflake.py]: Nice, I just removed it.

**Review by jpy-git — APPROVED** [review]
LGTM, well done @chwiese 🚀 🌟

**Pull request merged**